AIController.cs

```diff
@@ -3,30 +3,35 @@
 using System.Collections;
 using System.Collections.Generic;
 
+using BehaviorTree;
+
 public class AIController : MonoBehaviour
 {
-	public BehaviorTree behavior;
+	public BehaviorTree.BehaviorTree behavior;
 
-	private Hashtable data = new Hashtable();
+	private Hashtable _data = new Hashtable();
 
 	void Awake()
 	{
-		data["gameObject"] = gameObject;
+		_data["gameObject"] = gameObject;
+		behavior = BehaviorTreeEditor.CloneTree( behavior );
```

Comment:
It's a deep clone. Since all the nodes in the tree are assets simply calling `Instantiate()` on the behavior tree wouldn't clone all of the nodes, so I have to manually traverse the tree and clone each one.